CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -462,218 +758,330 @@ private void DestroyVertexBuffers() {
-        if ((maxNumberOfTexels > 0) && (numberOfTexels > maxNumberOfTexels)) {
-            CVKLOGGER.severe(String.format("CVKIconsRenderable.CreateXYZWTexture cannot allocate %d vertex points, maxium this device supports is %d", numberOfTexels, maxNumberOfTexels));
-            return VK_ERROR_TOO_MANY_OBJECTS;
-        }
-        
-        cvkXYZWTexelBuffer = CVKBuffer.Create(cvkDevice, 
-                                              cvkXYZWStagingBuffer.GetBufferSize(), 
+                
+        cvkPositionBuffer = CVKBuffer.Create(cvkDevice, 
+                                              cvkPositionStagingBuffer.GetBufferSize(), 
                                               VK_BUFFER_USAGE_UNIFORM_TEXEL_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
                                               VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
-        cvkXYZWStagingBuffer.DEBUGNAME = "CVKIconsRenderable.TaskCreateIcons cvkXYZWStagingBuffer";       
+        cvkPositionBuffer.DEBUGNAME = "CVKIconsRenderable.CreatePositionBuffer cvkPositionBuffer";       
                 
         try (MemoryStack stack = stackPush()) {
             // NB: we have already checked VK_FORMAT_R32G32B32A32_SFLOAT can be used as a texel buffer
             // format in CVKDevice.  If the format is changed here we need to check for its support in
             // CVKDevice.
             VkBufferViewCreateInfo vkViewInfo = VkBufferViewCreateInfo.callocStack(stack);
             vkViewInfo.sType(VK_STRUCTURE_TYPE_BUFFER_VIEW_CREATE_INFO);
-            vkViewInfo.buffer(cvkXYZWTexelBuffer.GetBufferHandle());
+            vkViewInfo.buffer(cvkPositionBuffer.GetBufferHandle());
             vkViewInfo.format(VK_FORMAT_R32G32B32A32_SFLOAT);
             vkViewInfo.offset(0);
             vkViewInfo.range(VK_WHOLE_SIZE);
 
             LongBuffer pBufferView = stack.mallocLong(1);
             ret = vkCreateBufferView(cvkDevice.GetDevice(), vkViewInfo, null, pBufferView);
             if (VkFailed(ret)) { return ret; }
-            hXYZWBufferView = pBufferView.get(0);
+            hPositionBufferView = pBufferView.get(0);
+            
+            // Descriptor sets will reference the old buffer view until updated
+            descriptorSetsState =  pDescriptorSets != null ? CVK_RESOURCE_NEEDS_UPDATE : CVK_RESOURCE_NEEDS_REBUILD;
         }       
         
-        ret = UpdateXYZWBuffer();
-        if (VkFailed(ret)) { return ret; }
-             
-        return ret;
+        return UpdatePositionBuffer();
     }
     
-    private int UpdateXYZWBuffer() {
+    private int UpdatePositionBuffer() {
         int ret;
         
-        ret = cvkXYZWTexelBuffer.CopyFrom(cvkXYZWStagingBuffer);       
-        cvkXYZWStagingBuffer.Destroy();
-        cvkXYZWStagingBuffer = null;
+        try {
+            positionStagingBufferLock.lock();
+            
+            List<DEBUG_CVKBufferElementDescriptor> DEBUG_positionDescriptors = new ArrayList<>();
```

Comment:
Should the debug stuff be in a Debug conditional?

Reply:
Not yet as we need to do some research on how to handle debug code.  I talked to two other Java devs on Constellation and they weren't aware of a good way to do this in Java as it's largely evaluated at runtime we can't #ifdef DEBUGGING the way we would in C/C++.